src/result.rs

```diff
@@ -70,71 +78,91 @@ pub enum CqlType {
 #[napi]
 impl QueryResultWrapper {
     pub fn from_query(internal: QueryResult) -> QueryResultWrapper {
-        QueryResultWrapper { internal }
+        let value = match internal.clone().into_rows_result() {
+            Ok(v) => QueryResultVariant::RowsResult(v),
+            Err(_) => QueryResultVariant::EmptyResult(internal),
+        };
+        QueryResultWrapper { internal: value }
     }
 
     #[napi]
-    pub fn get_rows(&self) -> Option<Vec<RowWrapper>> {
-        let rows = match &self.internal.rows {
-            Some(r) => r,
-            None => {
-                return None;
+    pub fn get_rows(&self) -> napi::Result<Option<Vec<RowWrapper>>> {
+        let r2 = match &self.internal {
+            QueryResultVariant::RowsResult(v) => v,
+            QueryResultVariant::EmptyResult(_) => {
+                return Ok(None);
             }
         };
 
-        Some(
-            rows.iter()
-                .map(|f| RowWrapper {
-                    internal: f.columns.clone(),
-                })
-                .collect(),
-        )
+        let rows = r2.rows::<Row>().map_err(err_to_napi)?.flatten();
```

Comment:
2. I would prefer to do it later, as this would require investigation how old driver behaves when it encounters that kind of problems.

Reply:
> I would prefer to do it later, as this would require investigation how old driver behaves when it encounters that kind of problems.

I insist on not allowing any errors to be silently ignored in our merged code.

Reply:
In this case I updated the code to panic in case on Deserialization error. The correct error handling I would still prefer to do at a later time.

Reply:
OK, I can accept this as a temporary measure.